DevSkim-DotNet/Microsoft.DevSkim/Ruleset.cs

```diff
@@ -184,34 +308,22 @@ public void AddString(string jsonstring, string sourcename, string? tag = null)
         /// </summary>
         /// <param name="languages"> Languages </param>
         /// <returns> Filtered rules </returns>
-        public IEnumerable<Rule> ByLanguages(string[] languages)
+        public IEnumerable<ConvertedOatRule> ByLanguages(string[] languages)
         {
-            // Otherwise preprare the rules for the content type and store it in cache.
-            List<Rule> filteredRules = new List<Rule>();
-
-            foreach (Rule r in _rules)
-            {
-                if (r.AppliesTo != null && ArrayContains(r.AppliesTo, languages))
-                {
-                    // Put rules with defined language (applies_to) on top
-                    filteredRules.Insert(0, r);
-                }
-                else if (r.AppliesTo == null || r.AppliesTo.Length == 0)
-                {
-                    // Put rules without applies_to on the bottom
-                    filteredRules.Add(r);
-                }
-            }
+            return _oatRules.Where(x => x.DevSkimRule.AppliesTo is null || x.DevSkimRule.AppliesTo.Length == 0 || (x.DevSkimRule.AppliesTo is string[] appliesList && appliesList.Any(y => languages.Contains(y))));
```

Comment:
This seems like a change in behavior as the old code was ordering the rules so that the more specific ones with a defined language scope would be evaluated first to improve the hit or capture quality.  The new code does not seem to maintain an ordering?

Reply:
got it